libpromises/cf3parse.y

```diff
@@ -483,10 +489,31 @@ promiser:              QSTRING
                                free(P.promiser);
                            }
                            P.promiser = P.currentstring;
+                           P.promiser_attribute = RvalNULL();
+
                            P.currentstring = NULL;
                            CURRENT_PROMISER_LINE = P.line_no;
                            ParserDebug("\tP:%s:%s:%s:%s:%s promiser = %s\n", P.block, P.blocktype, P.blockid, P.currenttype, P.currentclasses ? P.currentclasses : "any", P.promiser);
                        }
+
+                     | usefunction
+                       {
+                           if (P.promiser)
+                           {
+                               free(P.promiser);
+                           }
+
+                           Log(LOG_LEVEL_DEBUG, "Chasing promiser_attribute: fncall %s, nesting %d", P.currentfncall[P.arg_nesting+1]->name, P.arg_nesting+1);
+                           P.promiser_attribute = RvalNew(P.currentfncall[P.arg_nesting+1], RVAL_TYPE_FNCALL);
+                           Log(LOG_LEVEL_DEBUG, "Chasing promiser_attribute: resulting rval %s", RvalToString(P.promiser_attribute));
```

Comment:
This `RvalToString()` leaks. The same is true for a bunch of occurences of this function. What I would suggest would be to remove the "chasing" debug messages, they are confusing.